### PR TITLE
fix(perf-issues) ignore truncation queries and update thresholds for …

### PR DESCRIPTION
**Pull request by DominikB2014**
After auditing the first version of consecutive db queries, i've noticed we don't consider whether or not a query is truncated or not resulting in incorrect results from the dependance algorithm. This PR addresses this, as well as adds a duration threshold to the detector.

### CODE REVIEW

File: src/sentry/utils/performance_issues/performance_detection.py [hunk 1]
```diff
@@ -907,33 +911,48 @@ def _validate_and_store_performance_problem(self):
...
+            independant_db_spans
+        ) > self.settings.get("duration_threshold")
+        if exceeds_count_threshold and exceeds_duration_threshold:
             self._store_performance_problem()
 
     def _store_performance_problem(self) -> None:
         fingerprint = self._fingerprint()
+        offender_span_ids = list(
+            map(lambda span: span.get("span_id", None), self.consecutive_db_spans)
+        )
```
Comment (DominikB2014) [inline src/sentry/utils/performance_issues/performance_detection.py:925]: This should fix the issue where the `_pi_consecutive_db` tag is not set correctly.

File: .vscode/launch.json [hunk 2]
```diff
@@ -28,15 +28,6 @@
...
       "cwd": "${workspaceRoot}"
-    },
-    {
-      "name": "jest",
-      "type": "node",
-      "request": "launch",
-      "protocol": "inspector",
-      "program": "${workspaceFolder}/node_modules/.bin/jest",
-      "internalConsoleOptions": "openOnSessionStart",
-      "args": ["--runInBand"]
```
Comment (DominikB2014) [inline .vscode/launch.json:39]: Whoops, messed this up, going to revert this

File: src/sentry/utils/performance_issues/performance_detection.py [hunk 3]
```diff
@@ -907,33 +911,48 @@ def _validate_and_store_performance_problem(self):
...
+        independant_spans = []
         for span in spans[1:]:
             query: str = span.get("description", None)
-            if not query:
-                return False
-            if "WHERE" not in query.upper():  # TODO - use better regex
-                return True
-        return False
+            if (
+                query and contains_complete_query(span) and "WHERE" not in query.upper()
```
Comment (silent1mezzo) [inline src/sentry/utils/performance_issues/performance_detection.py:952]: Why do we care about complete queries for consecutive db spans? I can understand for N+1 where we need to ensure the query is the same.

Comment (DominikB2014) [reply src/sentry/utils/performance_issues/performance_detection.py:952]: With the current version, we need to ensure that the spans are independant of one another, requiring the full query.

File: src/sentry/utils/performance_issues/performance_detection.py [hunk 4]
```diff
@@ -452,12 +452,21 @@ def fingerprint_span_op(span: Span):
...
+def get_span_duration(span: Span) -> timedelta:
     return timedelta(seconds=span.get("timestamp", 0)) - timedelta(
         seconds=span.get("start_timestamp", 0)
     )
 
 
+def contains_complete_query(span: Span, is_source: Optional[bool] = False) -> bool:
+    # Remove the truncation check from the n_plus_one db detector.
+    query = span.get("description", None)
+    if is_source and query:
```
Comment (mjq-sentry) [inline src/sentry/utils/performance_issues/performance_detection.py:464]: I forgot about the `is_source` parameter on this - bit awkward now that this is pulled out of context but we can refactor later

File: src/sentry/utils/performance_issues/performance_detection.py [hunk 5]
```diff
@@ -907,33 +911,48 @@ def _validate_and_store_performance_problem(self):
         exceeds_count_threshold = len(self.consecutive_db_spans) >= self.settings.get(
             "consecutive_count_threshold"
         )
-        if exceeds_count_threshold and self._are_db_spans_independant(self.consecutive_db_spans):
+        independant_db_spans = self._find_independant_spans(self.consecutive_db_spans)
```
Comment (mjq-sentry) [inline src/sentry/utils/performance_issues/performance_detection.py:914]: spelling nit: `independent` (variable and method name)

File: src/sentry/utils/performance_issues/performance_detection.py (hunk 1 above)
Comment (mjq-sentry) [inline src/sentry/utils/performance_issues/performance_detection.py:925]: It seems like we generally prefer list comprehensions for this pattern, e.g.
```python
offender_span_ids = [span.get("span_id", None) for span in self.consecutive_db_spans)]
```

File: src/sentry/utils/performance_issues/performance_detection.py (hunk 5 above)
Comment (DominikB2014) [reply src/sentry/utils/performance_issues/performance_detection.py:914]: Fixed!

File: src/sentry/utils/performance_issues/performance_detection.py (hunk 1 above)
Comment (DominikB2014) [reply src/sentry/utils/performance_issues/performance_detection.py]: Yup, ive updated that, thanks! Thanks for the call out, simply was used to using map in python.

File: src/sentry/utils/performance_issues/performance_detection.py (hunk 4 above)
Comment (DominikB2014) [reply src/sentry/utils/performance_issues/performance_detection.py:464]: Will make a note of this

File: src/sentry/utils/performance_issues/performance_detection.py (hunk 5 above)
Comment (DominikB2014) [reply src/sentry/utils/performance_issues/performance_detection.py]: I actually make this mistake a lot lol

**Review by mjq-sentry — APPROVED** [review]
👍

**Pull request merged**